frontend/src/scenes/data-warehouse/editor/QueryTabs.tsx

```diff
@@ -34,19 +38,47 @@ interface QueryTabProps {
     onClick: (model: QueryTab) => void
     onClear?: (model: QueryTab) => void
     active: boolean
+    onRename: (model: QueryTab, newName: string) => void
 }
 
-function QueryTabComponent({ model, active, onClear, onClick }: QueryTabProps): JSX.Element {
+function QueryTabComponent({ model, active, onClear, onClick, onRename }: QueryTabProps): JSX.Element {
+    const [tabName, setTabName] = useState(() => model.name || NEW_QUERY)
+    const [isEditing, setIsEditing] = useState(false)
+
+    const handleRename = (): void => {
+        setIsEditing(false)
+        onRename(model, tabName)
+    }
+
     return (
-        <button
+        <div
             onClick={() => onClick?.(model)}
             className={clsx(
                 'space-y-px rounded-t p-1 flex flex-row items-center gap-1 hover:bg-[var(--bg-light)] cursor-pointer',
                 active ? 'bg-[var(--bg-light)] border-t border-l border-r' : 'bg-bg-3000',
                 onClear ? 'pl-3 pr-2' : 'px-3'
             )}
         >
-            {model.view?.name ?? 'New query'}
+            {isEditing ? (
+                <AutoTab
+                    value={tabName}
+                    onChange={(e) => setTabName(e.target.value)}
+                    onBlur={handleRename}
+                    autoFocus
+                    handleRename={() => onRename(model, tabName)}
+                    onKeyDown={(e) => {
+                        if (e.key === 'Enter') {
+                            handleRename()
+                        } else if (e.key === 'Escape') {
+                            setIsEditing(false)
+                        }
+                    }}
+                />
+            ) : (
+                <div onClick={() => setIsEditing(!isEditing)} className="flex-grow text-left whitespace-pre">
```

Comment:
Maybe make this something like dblclick? It's a bit too easy to end up in the editing mode (even when just navigating between tabs)